src/unix.rs

```diff
@@ -368,6 +392,73 @@ impl Client {
     }
 }
 
+#[cfg(any(target_os = "linux", target_os = "android"))]
+mod linux {
+    use super::*;
+
+    use libc::{iovec, off_t, ssize_t, syscall, SYS_preadv2};
+
+    // TODO: Replace this with libc::RWF_NOWAIT once they have it for musl
+    //  targets
+    const RWF_NOWAIT: c_int = 0x00000008;
+
+    fn cvt_ssize(t: ssize_t) -> io::Result<ssize_t> {
+        if t == -1 {
+            Err(io::Error::last_os_error())
+        } else {
+            Ok(t)
+        }
+    }
+
+    fn preadv2(fd: c_int, iov: &iovec) -> ssize_t {
+        let iovcnt: c_int = 1;
+        let offset: off_t = -1;
+
+        if cfg!(all(target_arch = "x86_64", target_pointer_width = "64")) {
```

Comment:
The branches here could use some comment. It's a bit surprising that 86-64 needs two special-cases but all other platforms (including powerpc, aarch64 and even  i686) all go in the same bucket.